sql/src/main/java/com/vmware/ddlog/util/sql/CalciteToH2.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2018-2020 VMware, Inc. All Rights Reserved.
+ * SPDX-License-Identifier: BSD-2
+ */
+
+package com.vmware.ddlog.util.sql;
+
+import org.apache.calcite.schema.ColumnStrategy;
+import org.apache.calcite.sql.SqlKind;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.SqlNodeList;
+import org.apache.calcite.sql.ddl.SqlColumnDeclaration;
+import org.apache.calcite.sql.ddl.SqlCreateTable;
+import org.apache.calcite.sql.util.SqlBasicVisitor;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+// We just need to strip out the subtype for the array, which H2 doesn't like.
+// Unfortunately this means we need to walk all columns in the parse tree.
+public class CalciteToH2 extends SqlBasicVisitor<String> {
+    @Override
+    public String visit(SqlNodeList nodeList) {
+        for (SqlNode statement : nodeList) {
+            if (statement.getKind() == SqlKind.CREATE_TABLE && statement instanceof SqlCreateTable) {
+                SqlCreateTable createStatement = (SqlCreateTable) statement;
+
+                List<SqlNode> operands = createStatement.getOperandList();
+                // For the SqlCreateTable class, getOperandList returns [name, columnList, query]
+
+                List<String> prestoColumns = new ArrayList();
+                if (operands.get(1) instanceof SqlNodeList) {
+                    SqlNodeList columns = (SqlNodeList) operands.get(1);
+                    for (SqlNode column : columns) {
+                        if (column instanceof SqlColumnDeclaration) {
```

Comment:
In general a visitor traverses recursively all nodes of an IR tree. I wonder whether you can just add this code to the visitor function for the column declaration instead.

Reply:
The visitor doesn't appear to have any logic that recursively walks an IR tree, but I refactored some of this code to call into that recursion (call `visit(SqlCall call)` inside `visit(SqlNodeList)`).

Also refactored the shared `visit(SqlNodeList)` code into a separate class.